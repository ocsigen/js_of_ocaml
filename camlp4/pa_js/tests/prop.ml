(* Test prop reading. *)

fun (obj : int) -> obj##p;;

fun (obj : < > Js.t) -> obj##p;;

fun (obj : < p : float Js.writeonly_prop > Js.t) -> obj##p + 1;;

fun (obj : < p : float Js.prop > Js.t) -> obj##p + 1;;
