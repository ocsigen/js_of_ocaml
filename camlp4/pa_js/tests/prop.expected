

#     Characters 45-48:
  fun (obj : int) -> obj##p;;
                     ^^^
Error: This expression has type int but an expression was expected of type
         < .. > Js.t
#   Characters 25-28:
  fun (obj : < > Js.t) -> obj##p;;
                          ^^^
Error: This expression has type <  >
       It has no method p
#   Characters 53-59:
  fun (obj : < p : float Js.writeonly_prop > Js.t) -> obj##p + 1;;
                                                      ^^^^^^
Error: This expression has type
         float Js.writeonly_prop = < set : float -> unit > Js.gen_prop
       but an expression was expected of type
         < get : 'jsoo_res; .. > Js.gen_prop
       The first object type has no method get
#   Characters 43-49:
  fun (obj : < p : float Js.prop > Js.t) -> obj##p + 1;;
                                            ^^^^^^
Error: This expression has type float but an expression was expected of type
         int
# 
